gax/src/test/java/com/google/api/gax/core/DistributionTest.java

```diff
@@ -29,66 +29,44 @@
  */
 package com.google.api.gax.core;
 
-import com.google.common.truth.Truth;
-import org.junit.Before;
+import static com.google.common.truth.Truth.assertThat;
+
+import java.util.Arrays;
+import java.util.Collections;
 import org.junit.Test;
 
 public class DistributionTest {
-  private Distribution distribution;
-
-  @Before
-  public void setup() {
-    distribution = new Distribution(10);
-  }
-
+  // These tests come from examples in https://en.wikipedia.org/wiki/Percentile#The_nearest-rank_method
   @Test
-  public void testGetCount() {
-    distribution.record(0);
-    distribution.record(1);
-    distribution.record(100);
-    Truth.assertThat(distribution.getCount()).isEqualTo(3);
-  }
+  public void testPercentile() {
+    Distribution dist;
 
-  @Test
-  public void testGetMean() {
-    distribution.record(0);
-    distribution.record(1);
-    Truth.assertThat(distribution.getMean()).isWithin(0.01).of(0.5);
+    dist = with(15, 20, 35, 40, 50);
+    assertThat(dist.getNthPercentile(5)).isEqualTo(15);
+    assertThat(dist.getNthPercentile(30)).isEqualTo(20);
+    assertThat(dist.getNthPercentile(40)).isEqualTo(20);
+    assertThat(dist.getNthPercentile(50)).isEqualTo(35);
+    assertThat(dist.getNthPercentile(100)).isEqualTo(50);
 
-    distribution.record(1);
-    Truth.assertThat(distribution.getMean()).isWithin(0.01).of(0.66);
+    dist = with(3, 6, 7, 8, 8, 10, 13, 15, 16, 20);
+    assertThat(dist.getNthPercentile(25)).isEqualTo(7);
+    assertThat(dist.getNthPercentile(50)).isEqualTo(8);
+    assertThat(dist.getNthPercentile(75)).isEqualTo(15);
+    assertThat(dist.getNthPercentile(100)).isEqualTo(20);
 
-    distribution.record(2);
-    Truth.assertThat(distribution.getMean()).isWithin(0.01).of(1);
+    dist = with(3, 6, 7, 8, 8, 9, 10, 13, 15, 16, 20);
+    assertThat(dist.getNthPercentile(25)).isEqualTo(7);
+    assertThat(dist.getNthPercentile(50)).isEqualTo(9);
+    assertThat(dist.getNthPercentile(75)).isEqualTo(15);
+    assertThat(dist.getNthPercentile(100)).isEqualTo(20);
   }
 
-  @Test
-  public void testReset() {
-    distribution.record(1);
-    distribution.reset();
-    Truth.assertThat(distribution.getCount()).isEqualTo(0);
-    Truth.assertThat(distribution.getMean()).isWithin(0.01).of(0);
-  }
-
-  @Test
-  public void testCopy() {
-    Distribution copy = distribution.copy();
-    copy.record(1);
-    Truth.assertThat(distribution.getCount()).isEqualTo(0);
-    Truth.assertThat(copy.getCount()).isEqualTo(1);
-
-    for (long l : distribution.getBucketCounts()) {
-      Truth.assertThat(l).isEqualTo(0);
+  private Distribution with(Integer... values) {
```

Comment:
Using the name `of` would be more consistent with our other surface.

Reply:
Done